### PR TITLE
fix: set task status to Overdue when expected_end_date passes

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
- [x] Fixes
- [x] Patches
- [ ] Test Cases(not required)

<h3>Expected Behaviour/Fixes:</h3> 
it sets status of task to overdue if expected end date of task has passed

### CODE REVIEW

File: erpnext/projects/doctype/task/task.py [hunk 1]
```diff
@@ -244,6 +244,8 @@ def update_status(self):
 			from datetime import datetime
 			if self.exp_end_date < datetime.now().date():
 				self.db_set('status', 'Overdue', update_modified=False)
+				for project in self.projects:
```
Comment (neilLasrado) [inline erpnext/projects/doctype/task/task.py:247]: This is incorrect. 

Only the Default Project status must be set as overdue.

**Review by neilLasrado — CHANGES_REQUESTED** [review]

File: erpnext/projects/doctype/task/task.py (hunk 1 above)
Comment (MyuddinKhatri) [reply erpnext/projects/doctype/task/task.py:247]: Did this way as the Expected end date set is for the task though it is in multiple projects and status shown in child table are of task.

Comment (sahil28297) [reply erpnext/projects/doctype/task/task.py:247]: @neilLasrado the expected end date is associated with tasks and not with project, looks good to me. Any reason why only the default project status must be updated?

Comment (vishdha) [reply erpnext/projects/doctype/task/task.py:247]: @sahil28297  Neil right here. it should update the status for the default project in the child table. Default Project status associated with the task only. 
let's take e.g for Sprint 1 someone's last updated status is a code review and the second sprint's last updated status is Testing.  if the date elapses then only the default project status needs to change

any way Task status is associated with default project only thats why if you updated default project status it updated task status

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/projects/doctype/task/task.py (hunk 1 above)
Comment (sahil28297) [reply erpnext/projects/doctype/task/task.py:247]: @MyuddinKhatri please make the necessary changes as suggested by vishal

**Comment by MyuddinKhatri** [comment]
@vishdha @sahil28297 i have updated PR, please check

**Review by vishdha — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**